File: src/main/java/org/zenbot/szolnok/timetable/batch/step/bus/reader/UrlResourceItemReader.java
package org.zenbot.szolnok.timetable.batch.step.bus.reader;

import lombok.extern.slf4j.Slf4j;
import org.springframework.batch.item.ItemReader;
import org.springframework.stereotype.Component;
import org.zenbot.szolnok.timetable.batch.step.readurls.StringResourcesInMemoryStorage;

@Slf4j
@Component
public class UrlResourceItemReader implements ItemReader<String> {

    private final StringResourcesInMemoryStorage stringResourcesInMemoryStorage;

    public UrlResourceItemReader(StringResourcesInMemoryStorage stringResourcesInMemoryStorage) {
        this.stringResourcesInMemoryStorage = stringResourcesInMemoryStorage;
    }

    @Override
    public String read() {
        log.info("Reading next item from [UrlResouceInMemoryStorage]");
        return stringResourcesInMemoryStorage.get();
    }
}

File: szolnok-timetable-batch-bus/src/main/kotlin/org/zenbot/szolnok/timetable/batch/bus/batch/step/bus/reader/UrlResourceItemReader.kt
package org.zenbot.szolnok.timetable.batch.bus.batch.step.bus.reader

import org.slf4j.LoggerFactory
import org.springframework.batch.item.ItemReader
import org.springframework.stereotype.Component
import org.zenbot.szolnok.timetable.batch.bus.batch.step.readurls.StringResourcesInMemoryStorage

@Component
class UrlResourceItemReader(private val stringResourcesInMemoryStorage: StringResourcesInMemoryStorage) : ItemReader<String> {

    private val log = LoggerFactory.getLogger(UrlResourceItemReader::class.java)

    override fun read(): String? {
        log.info("Reading next item from [UrlResouceInMemoryStorage]")
        return stringResourcesInMemoryStorage.get()
    }
}
